src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,59 @@
+package duke;
+
+import duke.task.Task;
+
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
+
+/**
+ * Storage class containing all the logic for loading and saving to file.
+ */
+public class Storage {
+    private String fileName;
+    private String folderPath;
+
+    /**
+     * Creates new Storage object.
+     *
+     * @param fileName Path to save file.
+     * @param folderPath Path to folder containing save file.
+     */
+    public Storage(String fileName, String folderPath) {
+        this.fileName = fileName;
+        this.folderPath = folderPath;
+    }
+
+    /**
+     * Loads text from save file.
+     *
+     * @return A list of strings containing lines in save file.
+     * @throws DukeException if save file is not found or cannot be parsed.
+     */
+    public List<String> load() throws DukeException {
+        try {
+            List<String> res = Files.readAllLines(Paths.get(fileName));
+            return res;
+        } catch (Exception e) {
+            throw(new DukeException("No saved data found!"));
+        }
+    }
+
+    /**
+     * Saves text containing task data into save file.
+     *
+     * @param tasks A list of strings containing lines to save.
+     * @throws Exception if save file cannot be written.
+     */
+    public void save(TaskList tasks) throws Exception {
```

Comment:
Since you threw a DukeException for `load`, you might want to consider throwing that for `save` as well!

Reply:
Good spot! Will handle the save function with DukeException in mind.